databricks/sdk/core.py

```diff
@@ -25,53 +19,19 @@
-        self._debug_truncate_bytes = cfg.debug_truncate_bytes if cfg.debug_truncate_bytes else 96
-        self._retry_timeout_seconds = cfg.retry_timeout_seconds if cfg.retry_timeout_seconds else 300
-        self._user_agent_base = cfg.user_agent
-        self._session = requests.Session()
-        self._session.auth = self._authenticate
-
-        # Number of urllib3 connection pools to cache before discarding the least
-        # recently used pool. Python requests default value is 10.
-        pool_connections = cfg.max_connection_pools
-        if pool_connections is None:
-            pool_connections = 20
-
-        # The maximum number of connections to save in the pool. Improves performance
-        # in multithreaded situations. For now, we're setting it to the same value
-        # as connection_pool_size.
-        pool_maxsize = cfg.max_connections_per_pool
-        if cfg.max_connections_per_pool is None:
-            pool_maxsize = pool_connections
-
-        # If pool_block is False, then more connections will are created,
-        # but not saved after the first use. Blocks when no free connections are available.
-        # urllib3 ensures that no more than pool_maxsize connections are used at a time.
-        # Prevents platform from flooding. By default, requests library doesn't block.
-        pool_block = True
-
-        # We don't use `max_retries` from HTTPAdapter to align with a more production-ready
-        # retry strategy established in the Databricks SDK for Go. See _is_retryable and
-        # @retried for more details.
-        http_adapter = HTTPAdapter(pool_connections=pool_connections,
-                                   pool_maxsize=pool_maxsize,
-                                   pool_block=pool_block)
-        self._session.mount("https://", http_adapter)
-
-        # Default to 60 seconds
-        self._http_timeout_seconds = cfg.http_timeout_seconds if cfg.http_timeout_seconds else 60
-
-        self._error_parser = _Parser(extra_error_customizers=[_AddDebugErrorCustomizer(cfg)])
+        self._api_client = _BaseClient(debug_truncate_bytes=cfg.debug_truncate_bytes,
+                                       retry_timeout_seconds=cfg.retry_timeout_seconds,
+                                       user_agent_base=cfg.user_agent,
+                                       header_factory=cfg.authenticate,
+                                       max_connection_pools=cfg.max_connection_pools,
+                                       max_connections_per_pool=cfg.max_connections_per_pool,
+                                       pool_block=True,
+                                       http_timeout_seconds=cfg.http_timeout_seconds,
+                                       extra_error_customizers=[_AddDebugErrorCustomizer(cfg)],
+                                       clock=cfg.clock)
 
     @property
     def account_id(self) -> str:
```

Comment:
What differentiates `account_id` and `is_account_client` from the rest of the things moved upwards? They're unused here so I'm assuming they are publicly facing, so I'm just curious on how to decide on which config attributes should be decoupled from `ApiClient` and which needs to stay.

Reply:
The way I thought about it: account_id and is_account_client are not inherently necessary for making API requests. In essence, they are really parameters of the config that happen to be in the ApiClient as well. The current architecture of the SDK doesn't expose the config directly to the *Api classes, so config-level attributes are "exposed" through the ApiClient class instead. However, fixing this would involve more invasive changes in the codegen process, which aren't necessary for now and will probably be revisited anyways with the next SDK version.

It doesn't seem like `is_account_client` is used within the SDK today, but it is publicly exposed. This is strongly coupled to the Config, because it requires knowledge of a host + logic to compute whether the provided host is an accounts host, which currently resides in the Config. In order to not break any callers, I left it as-is.

All of the other parameters are actually used inside of `_BaseClient` and affect its behavior, so they need to be included at that level.